pkg/cmd/grafana-cli/commands/install_command.go

```diff
@@ -107,12 +108,19 @@ func InstallPlugin(pluginName, version string, c utils.CommandLine) error {
 	logger.Infof("into: %v\n", pluginFolder)
 	logger.Info("\n")
 
-	content, err := c.ApiClient().DownloadFile(pluginName, pluginFolder, downloadURL, checksum)
+	// Create temp file for downloading zip file
+	tmpFile, err := ioutil.TempFile("", "*.zip")
+	if err != nil {
+		errutil.Wrap("Failed to create temporary file", err)
+	}
+	defer os.Remove(tmpFile.Name())
```

Comment:
Could you also add a `defer tmpFile.Close()` call before this, in order to close the file?

Reply:
I suppose you meant adding `defer tmpFile.Close()` *after* this, to have it being executed *before* the deletion, right?

I actually closed the file in function `DownloadFile`. I've just changed that. However, using only `defer tmpFile.Close()` hides the possible error thrown by `Close()`.
Cf https://www.joeshaw.org/dont-defer-close-on-writable-files/
So I added both `defer tmpFile.Close()` and `err = tmpFile.Close()`. Does that suit you?
In normal cases, the file will be closed in `err = tmpFile.Close()`; the second `Close()`, in the deferred call, will throw an error that will be ignored.

Reply:
@olivierlemasle Oh yeah, you're right, `defer tmpFile.Close()` should be on the line after this. I didn't recall that defers are executed in LIFO fashion. Thanks for catching that.

About catching the error from `tmpFile.Close`, I suggest we solve it with the technique of using a named error return value (`err`), and setting err with the error returned from `tmpFile.Close` if it's nil (call `tmpFile.Close` in a deferred closure).